### PR TITLE
remove xip.io dependency, using env vars instead

### PR DESCRIPTION
**Pull request by concaf**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,8 @@
-FROM openshift3/nodejs-010-rhel7
+FROM registry.access.redhat.com/openshift3/nodejs-010-rhel7
```
Comment (concaf) [inline Dockerfile:1]: This makes the image accessible from any box, not only CDK/ADB.

File: Dockerfile [hunk 2]
```diff
@@ -1,7 +1,8 @@
...
 
 ADD . /opt/app-root/src/
 
 EXPOSE 8080
 
-CMD /bin/bash -c 'npm start'
\ No newline at end of file
+CMD sed -e 's/HYSTRIXDASHBOARDURLPORT/'"$HOSTIP":"$HYSTRIX_DASHBOARD_PORT"'/; s/ZIPKINQUERYURLPORT/'"$HOSTIP"':'"$ZIPKIN_QUERY_PORT"'/' index.html.unprocessed > index.html && /bin/bash -c 'npm start'
```
Comment (concaf) [inline Dockerfile]: Replaces the variables set in `index.html.processed` with their values from the environment variables, and saves the output to `index.html` which is then served to the client. I'm sure there are better ways to do this, comments?

File: frontend.js [hunk 3]
```diff
@@ -21,6 +21,11 @@ var app = express();
 
 app.use('/', express.static('.'));
 
+app.get('/env', function (req, res) {
+  res.json({ env: process.env });
+});
+
+
```
Comment (concaf) [inline frontend.js:28]: Send the environment variables to the client.

File: index.html.unprocessed [hunk 4]
```diff
@@ -195,13 +195,14 @@ <h3 class="panel-title">Service Chaining</h3>
 			<!--  Tab Hystrix Dashaboard -->
 			<div role="tabpanel" class="tab-pane active" id="hystrix">
 				<iframe style="padding-top: 5vh; border: 0; width: 100%; height: 80vh; position: inherit;" id="hystrix-dashboard-iframe"
-					src="http://hystrix-dashboard-helloworld-msa.rhel-cdk.10.1.2.2.xip.io/monitor/monitor.html?stream=http%3A%2F%2Fturbine-server%2Fturbine.stream"> </iframe>
+					src="http://HYSTRIXDASHBOARDURLPORT/monitor/monitor.html?stream=http%3A%2F%2Fturbine-server%2Fturbine.stream">
+		</iframe>
```
Comment (concaf) [inline index.html.unprocessed:199]: These are the variables in the `index.html.unprocessed` which are replaced in the `CMD` directive of frontend's Dockerfile.

File: index.html.unprocessed [hunk 5]
```diff
@@ -195,13 +195,14 @@ <h3 class="panel-title">Service Chaining</h3>
...
 			</div>
 
 			<!--  Tab Hystrix Dashaboard -->
 			<div role="tabpanel" class="tab-pane active" id="zipkin">
 				<iframe style="padding-top: 5vh; border: 0; width: 100%; height: 80vh; position: inherit;" id="zipkin-dashboard-iframe"
-					src="http://zipkin-query-helloworld-msa.rhel-cdk.10.1.2.2.xip.io/"> </iframe>
+					src="http://ZIPKINQUERYURLPORT/"> </iframe>
```
Comment (concaf) [inline index.html.unprocessed:205]: These are the variables in the `index.html.unprocessed` which are replaced in the `CMD` directive of frontend's Dockerfile.
￼Add a line note

File: scripts/api-gateway.js [hunk 6]
```diff
@@ -13,7 +13,7 @@
 
 function api_gateway() {
         $.ajax({
-            url : 'http://api-gateway-helloworld-msa.rhel-cdk.10.1.2.2.xip.io/api',
+            url : 'http://' + env.HOSTIP + ':' + env.API_GATEWAY_PORT  + '/api',
```
Comment (concaf) [inline scripts/api-gateway.js:16]: Makes api-gateway rely on environment variables.

File: scripts/api-gateway.js [hunk 7]
```diff
@@ -32,5 +32,15 @@ function api_gateway() {
...
 $(document).ready(function() {
+
+    $.ajax({
+        url: document.URL + 'env',
+        success: function (result) {
+            env = result.env;
+        },
+        async: false
+    });
+
```
Comment (concaf) [inline scripts/api-gateway.js:43]: Gets environment variables from the server.

File: scripts/browser-client.js [hunk 8]
```diff
@@ -50,10 +60,19 @@ function browser_query() {
     }
     //Make the invocation
     for (var x = 0; x < services.length; x++) {
-        invoke_ajax(services[x].url, services[x].id)
-    }
+				invoke_ajax('http://' + env["HOSTIP"] + ':' + env[services[x].port] + services[x].url, services[x].id)
+}
```
Comment (concaf) [inline scripts/browser-client.js:64]: Makes requests based on the environment variables received from the server.

File: scripts/browser-client.js [hunk 9]
```diff
@@ -50,10 +60,19 @@ function browser_query() {
...
 $(document).ready(function() {
+    $.ajax({
+        url: document.URL + 'env',
+        success: function (result) {
+            env = result.env;
+        },
+        async: false
+    });
+
+
```
Comment (concaf) [inline scripts/browser-client.js:76]: Gets environment variables from the server.

File: scripts/service-chain.js [hunk 10]
```diff
@@ -13,7 +13,7 @@
 
 function chain_query() {
         $.ajax({
-            url : 'http://hello-helloworld-msa.rhel-cdk.10.1.2.2.xip.io/api/hello-chaining',
+            url : 'http://' + env.HOSTIP + ':' + env.HELLO_PORT  + '/api/hello-chaining', 
             cache : false,
```
Comment (concaf) [inline scripts/service-chain.js:17]: Makes service chaining not rely on xip.io

File: scripts/service-chain.js [hunk 11]
```diff
@@ -32,5 +32,14 @@ function chain_query() {
...
 $(document).ready(function() {
+ 
+   $.ajax({
+        url: document.URL + 'env',
+        success: function (result) {
+            env = result.env;
+        },
+        async: false
+    });
+
```
Comment (concaf) [inline scripts/service-chain.js:43]: Gets environment variables from the server.

**Comment by pradeepto** [comment]
@rafabene Hey Rafael, Can you please have a look at Shubham's PR and let us know what we can do to get this working? Thanks.

**Comment by rafabene** [comment]
I tried to run these changes using the original helloworld-msa in CDK and the POD went in CrashLoopBackOff state. I don't have time to debug the causes at this moment but I would be happy if you can take a look in this, please.

**Comment by surajssd** [comment]
This PR is works on docker-engine because it leverages the fact that docker-engine allows port bindings and thus you can talk to containers running in the VM.

But its failing or would not work with OpenShift because, OpenShift does not allow port bindings for someone to talk from outside, it only allows routes as a mechanism to do it.

**Comment by rafabene** [comment]
This PR should be updated according to the last commit

**Comment by rafabene** [comment]
This PR is outdated. Merged PR #4 instead.

**Pull request closed without merging**